src/game.cpp

```diff
@@ -12222,6 +12222,13 @@ void game::start_calendar()
                               + get_option<int>( "SEASON_LENGTH" ) * 1_days * scen->initial_season()
                               + 4 * get_option<int>( "SEASON_LENGTH" ) * 1_days * ( scen->initial_year() - 1 )
                               + 1_days * get_option<int>( "SPAWN_DELAY" );
+    if( calendar::start_of_game < calendar::start_of_cataclysm ) {
+        // Hotfix to prevent game start  from occuring before the cataclysm.
+        // Should be replaced with full refactor of the start date
+        calendar::start_of_game = calendar::start_of_cataclysm
+                                  + 1_hours * scen->initial_hour()
+                                  + 1_days * get_option<int>( "SPAWN_DELAY" );
+    }
```

Comment:
I feel that the addition of `1_days * get_option<int>( "SPAWN_DELAY" )` should be moved to after the `if` block.  Otherwise there's an odd discontinuity in `start_of_game` where increasing `SPAWN_DELAY` can actually make you start sooner.

Reply:
I can't think of any situation where larger `SPAWN_DELAY` can make things start sooner.

Reply:
Suppose that when `SPAWN_DELAY` = 1 week, `start_of_game` is one day earlier than `start_of_cataclysm`.  Then, inside your new `if` branch, `start_of_game` will be set to `start_of_cataclysm` plus one week.
Now if `SPAWN_DELAY` is set to eight days, `start_of_game` will equal `start_of_cataclysm`, and will not be changed by the new code, so it will remain equal.
In this case, increasing `SPAWN_DELAY` by one day has made the game start sooner by one week.

Reply:
Ok I see it now.
Fixed it.
There are other bigger problems not adressed in this fix still. Will leave them to the refactor pull request.